api/plugin.go

```diff
@@ -69,13 +68,59 @@ func (s *Server) SignPluginMessages(c echo.Context) error {
 		return fmt.Errorf("fail to unmarshal payroll policy, err: %w", err)
 	}
 
+	var payload v1.KeysignPayload
+	err = json.Unmarshal([]byte(req.Messages[0]), &payload)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal payload: %w", err)
+	}
+
+	//construct the transaction from the payload
+
+	parsedABI, err := abi.JSON(strings.NewReader(erc20ABI))
```

Comment:
I don't think we should leave the logic here , as this is only relevant to EVM chains(L1/L2)  , `v1.KeysignPayload` it could be represent any transactions